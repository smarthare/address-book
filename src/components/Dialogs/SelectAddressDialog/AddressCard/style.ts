import Grid from "@mui/material/Grid";
import { styled } from "@mui/material/styles";

export const StyledAddressCard = styled(Grid, {
  shouldForwardProp: (prop) => prop !== "selected",
})(({ selected }: { selected?: any }) => ({
  cursor: "pointer",
  padding: "10px",

  "&>div": {
    ...(selected && {
      background: "#333",
      "& *": {
        color: "#fff",
      },
    }),
    border: "1px solid #D9D9D9",
    borderRadius: "10px",
    padding: "16px",
  },
}));

export const StyledAddressCardHeader = styled("div")(() => ({
  display: "flex",
  justifyContent: "space-between",
  marginBottom: "16px",
}));
