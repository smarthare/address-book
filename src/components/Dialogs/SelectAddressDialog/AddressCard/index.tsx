import { useContext } from "react";
import { Typography } from "@mui/material";

import { CheckIcon, HomeIcon } from "components/icons";

import { AddressContext } from "contexts/AddressContext";
import { AddressCardProps } from "types/props";
import { StyledAddressCard, StyledAddressCardHeader } from "./style";
import { StyledButton } from "../style";

const AddressCard = (props: AddressCardProps) => {
  const { addressId, texts, selected, newCard, handleClick } = props;

  const { setCurrentId } = useContext(AddressContext);

  const handleClickSetWrapper = () => {
    newCard ? handleClick() : addressId && setCurrentId(addressId);
  };

  return (
    <StyledAddressCard
      xs={12}
      sm={12}
      md={6}
      lg={4}
      item
      selected={selected}
      onClick={handleClickSetWrapper}
    >
      <div>
        <StyledAddressCardHeader>
          <HomeIcon />
          {!newCard && selected && <CheckIcon />}
        </StyledAddressCardHeader>

        {texts.map(
          (text, ind) =>
            ind !== 2 && (
              <Typography
                variant="h3"
                fontSize="1.5rem"
                fontWeight="bold"
                gutterBottom
                key={`address-${ind}-${text}`}
              >
                {text}
              </Typography>
            )
        )}

        {!newCard && <StyledButton onClick={handleClick}>Edit</StyledButton>}
      </div>
    </StyledAddressCard>
  );
};

export default AddressCard;
