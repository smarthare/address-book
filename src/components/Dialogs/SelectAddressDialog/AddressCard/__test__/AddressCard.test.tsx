import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

import AddressCard from "..";

afterEach(() => {
  cleanup();
});

test("Should render AddressCard component", () => {
  render(
    <AddressCard
      addressId={""}
      texts={["test1", "skip", "test2"]}
      selected={true}
      handleClick={() => {}}
    />
  );

  const cardElement = screen.getByTestId("cardAddress");

  expect(cardElement).toBeInTheDocument();
  expect(cardElement).toHaveClass("MuiGrid-root");
  expect(cardElement.querySelectorAll("h3").length).toEqual(2);
  expect(screen.getByTestId("cardBtn")).toBeInTheDocument();
});

test("AddressCard matches snapshot", () => {
  const component = render(
    <AddressCard
      addressId={"test-address-id"}
      texts={["test1", "test2", "test3"]}
      selected={true}
      handleClick={() => {}}
    />
  );

  expect(component).toMatchSnapshot();
});
