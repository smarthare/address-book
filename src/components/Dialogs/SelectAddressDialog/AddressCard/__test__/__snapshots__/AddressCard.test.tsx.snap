// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressCard matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-1q6dfag-MuiGrid-root"
        data-testid="cardAddress"
      >
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <div
            class="css-1abudc7"
          >
            <div
              class="css-ed9sft"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ev5wqq-MuiSvgIcon-root"
              data-testid="CheckCircleRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM9.29 16.29 5.7 12.7a.9959.9959 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0L10 14.17l6.88-6.88c.39-.39 1.02-.39 1.41 0 .39.39.39 1.02 0 1.41l-7.59 7.59c-.38.39-1.02.39-1.41 0z"
              />
            </svg>
          </div>
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-1lp8xxv-MuiTypography-root"
          >
            test1
          </h3>
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-1lp8xxv-MuiTypography-root"
          >
            test3
          </h3>
          <div
            class="MuiGrid-root MuiGrid-container css-79zp5u-MuiGrid-root"
          >
            <button
              class="css-123zum0"
              data-testid="cardBtn"
            >
              Edit
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-1q6dfag-MuiGrid-root"
      data-testid="cardAddress"
    >
      <div
        class="MuiGrid-root css-vj1n65-MuiGrid-root"
      >
        <div
          class="css-1abudc7"
        >
          <div
            class="css-ed9sft"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
          </div>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ev5wqq-MuiSvgIcon-root"
            data-testid="CheckCircleRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM9.29 16.29 5.7 12.7a.9959.9959 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0L10 14.17l6.88-6.88c.39-.39 1.02-.39 1.41 0 .39.39.39 1.02 0 1.41l-7.59 7.59c-.38.39-1.02.39-1.41 0z"
            />
          </svg>
        </div>
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-1lp8xxv-MuiTypography-root"
        >
          test1
        </h3>
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-1lp8xxv-MuiTypography-root"
        >
          test3
        </h3>
        <div
          class="MuiGrid-root MuiGrid-container css-79zp5u-MuiGrid-root"
        >
          <button
            class="css-123zum0"
            data-testid="cardBtn"
          >
            Edit
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
